psi4/src/psi4/libfock/DirectJK.cc

```diff
@@ -271,52 +363,83 @@ void DirectJK::compute_JK() {
     }
 #endif
 
+    if (incfock_) {
+        timer_on("DirectJK: INCFOCK Preprocessing");
+        incfock_setup();
+        int reset = options_.get_int("INCFOCK_FULL_FOCK_EVERY");
+        double dconv = options_.get_double("D_CONVERGENCE");
+        double Dnorm = Process::environment.globals["SCF D NORM"];
+        // Do IFB on this iteration?
+        do_incfock_iter_ = (Dnorm >= dconv) && !initial_iteration_ && (incfock_count_ % reset != reset - 1);
+        
+        if (!initial_iteration_ && (Dnorm >= dconv)) incfock_count_ += 1;
+        timer_off("DirectJK: INCFOCK Preprocessing");
+    }
+
     auto factory = std::make_shared<IntegralFactory>(primary_, primary_, primary_, primary_);
+    
+    std::vector<SharedMatrix>& D_ref = (do_incfock_iter_ ? delta_D_ao_ : D_ao_);
+    std::vector<SharedMatrix>& J_ref = (do_incfock_iter_ ? delta_J_ao_ : J_ao_);
+    std::vector<SharedMatrix>& K_ref = (do_incfock_iter_ ? delta_K_ao_ : K_ao_);
+    std::vector<SharedMatrix>& wK_ref = (do_incfock_iter_ ? delta_wK_ao_ : wK_ao_);
 
     if (do_wK_) {
         std::vector<std::shared_ptr<TwoBodyAOInt>> ints;
         for (int thread = 0; thread < df_ints_num_threads_; thread++) {
             ints.push_back(std::shared_ptr<TwoBodyAOInt>(factory->erf_eri(omega_)));
+            if (density_screening_) ints[thread]->update_density(D_ref);
         }
```

Comment:
If this shows up in the timings for parallel runs, we can change it to building a single integral object, then cloning it for all threads with an ID greater than 0.  It's correct as-is, so I don't suggest we change it unless we see L2 object construction times becoming significant.

Reply:
iirc, update density is a REALLY cheap step.

Reply:
`Update Density                      :      6.883u      0.050s      0.234w      9 calls`

Cheapness confirmed, this is on an 18 core machine.